### ISSUE TITLE
Support for etcd 3

### ISSUE THREAD

**Issue by hseeberger** [hseeberger/constructr#124]

**Comment by Chumper** [comment]
etcd3 should be straitght forward to implement. I used the grpc api on my own adapter: https://github.com/Chumper/etcd3-scala
It is far from perfect, I use it to learn scala along the way but I hope you can use it as inspiration for an etcd3 version.

I can also try to create an etcd3 version as plugin if you want.

**Comment by hseeberger** [comment]
Thanks, but I'm already on it, thereby changing some semantics of ConstructR ;-)

**Comment by Chumper** [comment]
Yeah, afterwards I saw your etcd3 branch that you recently started. Good work btw.

**Comment by yi719** [comment]
Hi, are there any plans on releasing Support for etcd 3 ?

**Comment by lutzh** [comment]
I set up a little test system with constructr and unaware of etcd3 support being w.i.p., I naively used etcd3. And as far as I can tell it's working just fine !?!

Is my understanding correct, that the etcd3 support means making use of the gRPC interface of etcd3, but even without, constructr 0.17 works fine with etcd3 (because of etcd3's Json backwards compatibility)?

What problems should I expect when using constructr 0.17 with etcd3?
